AssertionError: Type name cannot be empty

```
ERROR: While parsing lib/OpenCloud/Database/Resource/Configuration.php...
AssertionError: Type name cannot be empty in /home/jim/projects/phan/src/Phan/Language/Type.php:422
Stack trace:
#0 /home/jim/projects/phan/src/Phan/Language/Type.php(1392): Phan\Language\Type::make('\\', '', Array, false, 2)
#1 /home/jim/projects/phan/src/Phan/Language/UnionType.php(385): Phan\Language\Type::fromStringInContext('\\', Object(Phan\Language\Context), 2, Object(Phan\CodeBase))
#2 /home/jim/projects/phan/src/Phan/Language/Element/Comment/Builder.php(259): Phan\Language\UnionType::fromStringInContext('\\', Object(Phan\Language\Context), 2, Object(Phan\CodeBase))
#3 /home/jim/projects/phan/src/Phan/Language/Element/Comment/Builder.php(633): Phan\Language\Element\Comment\Builder->returnTypeFromCommentLine('* @return \\@std...', 3)
#4 /home/jim/projects/phan/src/Phan/Language/Element/Comment/Builder.php(393): Phan\Language\Element\Comment\Builder->maybeParseReturn(3, '* @return \\@std...')
#5 /home/jim/projects/phan/src/Phan/Language/Element/Comment/Builder.php(320): Phan\Language\Element\Comment\Builder->parseCommentLine(3, '* @return \\@std...')
#6 /home/jim/projects/phan/src/Phan/Language/Element/Comment.php(411): Phan\Language\Element\Comment\Builder->build()
#7 /home/jim/projects/phan/src/Phan/Language/Element/Method.php(515): Phan\Language\Element\Comment::fromStringInContext('/**\n     * Gene...', Object(Phan\CodeBase), Object(Phan\Language\Context), 68, 5)
#8 /home/jim/projects/phan/src/Phan/Parse/ParseVisitor.php(314): Phan\Language\Element\Method::fromNode(Object(Phan\Language\Context), Object(Phan\CodeBase), Object(ast\Node), Object(Phan\Language\FQSEN\FullyQualifiedMethodName), Object(Phan\Language\Element\Clazz))
#9 /home/jim/projects/phan/src/Phan/Analysis.php(187): Phan\Parse\ParseVisitor->visitMethod(Object(ast\Node))
#10 /home/jim/projects/phan/src/Phan/Analysis.php(214): Phan\Analysis::parseNodeInContext(Object(Phan\CodeBase), Object(Phan\Language\Context), Object(ast\Node))
#11 /home/jim/projects/phan/src/Phan/Analysis.php(214): Phan\Analysis::parseNodeInContext(Object(Phan\CodeBase), Object(Phan\Language\Context), Object(ast\Node))
#12 /home/jim/projects/phan/src/Phan/Analysis.php(214): Phan\Analysis::parseNodeInContext(Object(Phan\CodeBase), Object(Phan\Language\Context), Object(ast\Node))
#13 /home/jim/projects/phan/src/Phan/Analysis.php(142): Phan\Analysis::parseNodeInContext(Object(Phan\CodeBase), Object(Phan\Language\Context), Object(ast\Node))
#14 /home/jim/projects/phan/src/Phan/Phan.php(207): Phan\Analysis::parseFile(Object(Phan\CodeBase), 'lib/OpenCloud/D...')
#15 /home/jim/projects/phan/src/phan.php(34): Phan\Phan::analyzeFileList(Object(Phan\CodeBase), Object(Closure))
#16 /home/jim/projects/phan/phan(2): require_once('/home/jim/proje...')
#17 {main}
```

steps to reproduce:
`git clone git@github.com:rackspace/php-opencloud.git && cd php-opencloud && composer install --no-dev && phan --init`
`phan`

phan version:
- Phan 2.4.4
- Phan 2.4.5-dev

--- a/NEWS.md
+++ b/NEWS.md
@@ -8,7 +8,7 @@ Plugins:
   make it affect all methods of descendant classlikes that inherited that method definition.
 
 New Features(Analysis)
-+ Infer that `!empty($x['field']...)` also implies $x['field'] is non-falsey. (#3570)
++ Infer that `!empty($x['field']...)` also implies `$x['field']` is non-falsey. (#3570)
 
 Bug fixes:
 + Fix bug in native parsing of `AST_TYPE_UNION` (union type) nodes for PHP 8.0.0-dev.
@@ -18,6 +18,7 @@ Bug fixes:
   `PASSWORD_DEFAULT` became null in php 7.4, and other constants became strings.
 
   Note that you will need to run Phan with both php 7.4 and a `target_php_version` of 7.4 to fix the errors.
++ Fix uncaught `AssertionError` when parsing `@return \\...` (#3573)
 
 Nov 24 2019, Phan 2.4.4
 -----------------------

--- a/src/Phan/Language/Type.php
+++ b/src/Phan/Language/Type.php
@@ -105,7 +105,7 @@ class Type
      * A legal type identifier (e.g. 'int' or 'DateTime')
      */
     const simple_type_regex_or_this =
-        '(\??)(callable-(?:string|object|array)|associative-array|class-string|non-empty-(?:associative-array|array|list)|[a-zA-Z_\x7f-\xff\\\][a-zA-Z0-9_\x7f-\xff\\\]*|\$this)';
+        '(\??)(callable-(?:string|object|array)|associative-array|class-string|non-empty-(?:associative-array|array|list)|\\\\?[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*(?:\\\\[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*)*|\$this)';
 
     const shape_key_regex =
         '(?:[-.\/^;$%*+_a-zA-Z0-9\x7f-\xff]|\\\\(?:[nrt\\\\]|x[0-9a-fA-F]{2}))+\??';

--- a/tests/files/expected/0828_crash_return_backslashes.php.expected
+++ b/tests/files/expected/0828_crash_return_backslashes.php.expected
@@ -1,0 +1,1 @@
+%s:4 PhanUnextractableAnnotation Saw unextractable annotation for comment '* @return \\@Test'

--- a/tests/files/src/0828_crash_return_backslashes.php
+++ b/tests/files/src/0828_crash_return_backslashes.php
@@ -1,0 +1,7 @@
+<?php
+
+/**
+ * @return \\@Test
+ */
+function test() {
+}

